docs/06/01/types.md

```diff
@@ -0,0 +1,159 @@
+## Component Adapter Types
+
+The Component Adapter system uses TypeScript interfaces to ensure type safety and consistent behavior. Understanding these types is essential for creating custom component implementations that work correctly with the SDK.
+
+### ComponentsContextType
+
+This interface defines the shape of the component adapter object. It specifies every UI component that can be customized in the SDK ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/useComponentContext.ts)):
+
+```typescript
+export interface ComponentsContextType {
+  Alert: (props: AlertProps) => JSX.Element | null
+  Badge: (props: BadgeProps) => JSX.Element | null
+  Button: (props: ButtonProps) => JSX.Element | null
+  ButtonIcon: (props: ButtonIconProps) => JSX.Element | null
+  CalendarPreview: (props: CalendarPreviewProps) => JSX.Element | null
+  Card: (props: CardProps) => JSX.Element | null
+  Checkbox: (props: CheckboxProps) => JSX.Element | null
+  CheckboxGroup: (props: CheckboxGroupProps) => JSX.Element | null
+  ComboBox: (props: ComboBoxProps) => JSX.Element | null
+  DatePicker: (props: DatePickerProps) => JSX.Element | null
+  Heading: (props: HeadingProps) => JSX.Element | null
+  Link: (props: LinkProps) => JSX.Element | null
+  Menu: (props: MenuProps) => JSX.Element | null
+  NumberInput: (props: NumberInputProps) => JSX.Element | null
+  OrderedList: (props: OrderedListProps) => JSX.Element | null
+  PaginationControl?: (props: PaginationControlProps) => JSX.Element | null
+  ProgressBar: (props: ProgressBarProps) => JSX.Element | null
+  Radio: (props: RadioProps) => JSX.Element | null
+  RadioGroup: (props: RadioGroupProps) => JSX.Element | null
+  Select: (props: SelectProps) => JSX.Element | null
+  Switch: (props: SwitchProps) => JSX.Element | null
+  Table: <T>(props: TableProps<T>) => JSX.Element | null
+  Text: (props: TextProps) => JSX.Element | null
+  TextInput: (props: TextInputProps) => JSX.Element | null
+  UnorderedList: (props: UnorderedListProps) => JSX.Element | null
+}
+```
+
+Each component type (e.g., `ButtonProps`, `TextInputProps`) is defined in its respective file and exported from the SDK. These interfaces detail all the props that must be supported by your custom implementations.
+
+### GustoProviderCustomUIAdapterProps
```

Comment:
This potentially seems out of place in this doc? I'm not opposed to having it but it may belong in a getting started section where the GustoProvider is already explained?

Feeling also like we should favor just linking to the interface so that it can stay up to date rather than enumerating the values here since the interface is likely to get out of date quickly

Reply:
Updated, also rewrote the types file to push more stuff into the setup-guide which i agree makes more sense.